src/hypothesis/stateful.py

```diff
@@ -518,3 +575,8 @@ def execute_step(self, step):
             )
             for target in rule.targets:
                 self.bundle(target).append(VarReference(name))
+
```

Comment:
Question: Currently invariants run after every rule, but this means that the invariant has not been checked before the first rule runs.

Is this desirable? Would it make sense to check invariants immediately and fail the test if they are not satisfied from the outset? The main benefit to doing so I can think of is that it would result in faster failures when there are trivial errors in the invariants, but given that stateful testing can be quite slow (though it probably shouldn't be in the case that only one rule fires) this might not be a small benefit.